pkg/objstore/azure/helpers.go

```diff
@@ -30,20 +34,57 @@ func init() {
 	pipeline.SetForceLogEnabled(false)
 }
 
+func getAzureStorageCredentials(conf Config) (blob.Credential, error) {
+
+	if conf.MSIResource != "" {
+
+		msiConfig := auth.NewMSIConfig()
+		msiConfig.Resource = conf.MSIResource
+
+		azureServicePrincipalToken, err := msiConfig.ServicePrincipalToken()
+		if err != nil {
+			return nil, err
+		}
+
+		// Get a new token
+		err = azureServicePrincipalToken.Refresh()
+		if err != nil {
+			return nil, err
+		}
+		token := azureServicePrincipalToken.Token()
+
+		return blob.NewTokenCredential(token.AccessToken, nil), nil
+
+	} else {
+		credential, err := blob.NewSharedKeyCredential(conf.StorageAccountName, conf.StorageAccountKey)
+		if err != nil {
+			return nil, err
+		}
+		return credential, nil
+	}
+
```

Comment:
```suggestion
```

Reply:
Removed the `else` so this newline is now legit.